spinoso-string/src/lib.rs

```diff
@@ -272,21 +114,21 @@ impl PartialEq for String {
         // [3.0.2] > s == t
         // => true
         // ```
-        self.buf[..] == other.buf[..]
+        *self.inner.as_slice() == *other.inner.as_slice()
```

Comment:
So, i was doubting whether this is correct, but i'm 99% sure it is. You were previously doing direct slice comparison, and since `as_slice` returns a ref, then we needed to deref it.

However, since the encoded types now support Deref (returns `&[u8]`, we could make this `*self.inner == *other.inner` right?

Reply:
either works, but I'd prefer to push this down into `EncodedString`. The impl in `EncodedString` can basically be `*self.as_slice() == *other.has_slice()`.

The reason to do the deref on the slice is to use `PartialEq<[u8]> for [u8]` impl rather than an extra indirection with an additional ref.

Reply:
I think I mostly follow what you mean :grimacing:. By pushing this down to `EncodedString`, are you meaning `impl PartialEq for EncodedString` (e.g. not on the layer further down right?) and then just using the redirected `as_slice()` methods?

Just checking that last part. if we did `self.as_slice() == other.as_slice()`, since those `as_slice()` are `&[u8]`, then it would use `impl PartialEq<&[u8]> for AsciiString` (in asciiString for example), which in itself uses `**self` instead of `*self`? (I'm realising `Partial<[u8]> for AsciiString` might have a typo, it is currently `**self = *other` instead of `*self = *other`)